### DISCUSSION TITLE
dulwich.porcelain.clone No such file or directory on AWS Lambda

### DISCUSSION THREAD

**Discussion by j2ali** [jelmer/dulwich#941] [Q&A] [answered]
I am trying to execute

```
dulwich.porcelain.clone('git@bitbucket.org:%s/%s.git' % (BITBUCKET_USER,BITBUCKET_REPO_NAME), '/tmp/repo')
```

in AWS Lambda, however I get 

```
"errorType": "OSError",
  "errorMessage": "[Errno 2] No such file or directory"
```

This works for me locally. Anyone having the same problem?

**Comment by jelmer** [comment]
Can you paste the full traceback? Apparently the /tmp on AWS Lambda is special in some way, though I don't have any experience with it myself.

**Comment by jelmer** [comment]
Closing this since it's still waiting for a reply after two years.

**Comment by jelmer** [comment, marked as answer]
Actually, I'm pretty sure this is because you don't have SSH installed. 

You can either install SSH or use the Paramiko SSH Vendor that lives in dulwich.contrib.paramiko.
